pkg/kexec/kexec_linux_arm.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2015-2019 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package kexec
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"github.com/u-root/u-root/pkg/dt"
+	"github.com/u-root/u-root/pkg/zimage"
+	"golang.org/x/sys/unix"
+)
+
+const (
+	maxCommandLineSize = 1023 // 1024 with the null character
+	stackSize          = 0x1000
+	heapSize           = 0x10000
+	textOffset         = 0x8000
+)
+
+// FileLoad loads the given kernel as the new kernel with the given options.
+//
+// For arm, this uses the kexec_load(2) syscall.
+func (opts *LinuxOpts) FileLoad(kernel *os.File) error {
```

Comment:
maybe leave these on their own, and look at pkg/boot.LinuxImage? I'm not sure what goes where, just making you aware